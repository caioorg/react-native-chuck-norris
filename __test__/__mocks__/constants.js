export const LIST_CATEGORY = [
  'animal',
  'career',
  'celebrity',
  'dev',
  'explicit',
  'fashion',
  'food',
  'history',
  'money',
  'movie',
  'music',
  'political',
  'religion',
  'sport',
  'travel',
];

export const JOKE_SELECTED = {
  value:
    'All arrays Chuck Norris declares are of infinite size, because Chuck Norris knows no bounds.',
  categories: ['dev'],
};
