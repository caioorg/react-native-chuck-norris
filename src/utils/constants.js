export const URLS = {
	CATEGORIES: 'https://api.chucknorris.io/jokes/categories',
	RANDOM_CATEGORY_FACT: 'https://api.chucknorris.io/jokes/random?category=',
};
