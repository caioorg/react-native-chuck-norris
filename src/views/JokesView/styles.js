import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0 20px;
  position: relative;
`;

export const JokeText = styled.Text`
  font-size: 20px;
  font-style: italic;
  text-align: center;
`;

export const JokeRefresh = styled.TouchableOpacity`
  padding: 20px 10px;
  position: absolute;
  bottom: 40px;
  right: 20px;
  background: #7b5e57;
  border-radius: 50px;
  height: 60px;
  width: 150px;
`;

export const JokeRefreshText = styled.Text`
  font-size: 16px;
  color: #fff;
  text-align: center;
  font-weight: bold;
`;
