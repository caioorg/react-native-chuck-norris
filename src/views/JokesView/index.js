import React from 'react';
import PropTypes from 'prop-types';
import {Container, JokeText, JokeRefresh, JokeRefreshText} from './styles';

const JokesView = ({data: {value, categories}, refresh}) => {
  return (
    <Container>
      {value && <JokeText>{value}</JokeText>}
      <JokeRefresh onPress={() => refresh(categories[0])}>
        <JokeRefreshText>Atualizar</JokeRefreshText>
      </JokeRefresh>
    </Container>
  );
};

JokesView.propTypes = {
  data: PropTypes.object.isRequired,
  refresh: PropTypes.func.isRequired,
};

export default JokesView;
