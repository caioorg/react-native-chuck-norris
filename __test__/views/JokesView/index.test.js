import React from 'react';
import renderer from 'react-test-renderer';
import JokesView from '../../../src/views/JokesView';
import {JOKE_SELECTED} from '../../__mocks__/constants';

const props = {data: JOKE_SELECTED, refresh: jest.fn()};

describe('[Views] Jokes', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<JokesView {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
