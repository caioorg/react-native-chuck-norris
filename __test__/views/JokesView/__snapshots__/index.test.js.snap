// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
        "position": "relative",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
          "fontStyle": "italic",
          "textAlign": "center",
        },
      ]
    }
  >
    All arrays Chuck Norris declares are of infinite size, because Chuck Norris knows no bounds.
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#7b5e57",
        "borderRadius": 50,
        "bottom": 40,
        "height": 60,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 20,
        "position": "absolute",
        "right": 20,
        "width": 150,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    >
      Atualizar
    </Text>
  </View>
</View>
`;
